lib/rubimas/idol.rb

```diff
@@ -50,7 +50,18 @@ def find(idol_id)
       alias_method :find_by_id, :find
 
       def find_by_name(idol_name)
-        all_idols.find { |idol| idol.key == idol_name } || raise("unknown idol: #{idol_name}")
+        case idol_name
+        when Symbol
+          all_idols.find { |idol| idol.key == idol_name } || raise("unknown idol: #{idol_name}")
+        when String
```

Comment:
How about the code below? I think we can decrease code volume.

```ruby
all_idols.find do |idol|
  [idol.key, idol.name.given, idol.name.aka, idol.name.full].include?(idol_name) || raise("unknown idol: #{idol_name}")
end
```

or we can extend `Idol::Name#==` method like below

```ruby
def ==(other)
  super || [aka, family, full].compact.include?(other)
end
```

and can write `find_by_name` method like this.

```ruby
all_idols.find { |idol| [idol.key, idol.name].include? idol_name } || raise("unknown idol: #{idol_name}")
```

How do you think?